cmds/core/yes/yes_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bufio"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/testutil"
+)
+
+func TestMain(m *testing.M) {
+	testutil.Run(m, main)
+}
+
+func TestYes(t *testing.T) {
+	type test struct {
+		in       []string
+		expected string
+	}
+	tests := []test{
+		{in: []string{}, expected: "y"},
+		{in: []string{"hi"}, expected: "hi"},
+		{in: []string{"hi", "how", "are", "you"}, expected: "hi how are you"},
+	}
+	for _, v := range tests {
+		cmd := testutil.Command(t, v.in...)
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			t.Fatalf("Failed to get stdout of yes command: got %v, want nil", err)
+		}
+
+		if err := cmd.Start(); err != nil {
+			t.Fatalf("Failed to start yes command: got %v, want nil", err)
+		}
+		in := bufio.NewScanner(stdout)
+		for i := 0; i < 1000; i++ {
+			if !in.Scan() {
+				t.Fatalf("Could not scan: got %v, want nil", in.Err())
+			}
+			if text := in.Text(); text != v.expected {
+				t.Errorf("Got %v at iteration %d, want %v", text, i, v.expected)
+				break
+			}
+		}
```

Comment:
This is just about there. 

There's only one problem: it is unusual for yes to get killed; more common for it to exit when the pipe it is writing to closes. We're not testing that latter scenario. One way to test it:
run each test twice, first time, close the stdout pipe and make sure it exits, second time, kill it and make sure it exits.

Another way is to add an element to the struct called
closeorkill bool

and depending on the value of close or kill, either close stdout or kill the child.

Reply:
Perfect. As discussed, there are three implementations. One is to run the test twice. Two is to add an element to the struct called closeOrKill bool. Three is to write an entire separate test function and changing the cmd.Process.Kill() to closing stdout pipe and then cmd.Close(). The first implementation would cause me to have three for loops. The third implementation causes me to repeat a lot of the same code, so I decided to go with the second implementation instead. :)